src/Akeneo/Pim/Enrichment/Bundle/Command/BackoffElasticSearchStateHandler.php

```diff
@@ -1,60 +1,76 @@
 use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\HttpFoundation\Response;
 
+/**
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
 class BackoffElasticSearchStateHandler
 {
-    public const RETRY_COUNTER = 10;
-    public const INITIAL_WAIT_DELAY = 10;
-    public const BACKOFF_LOGARITHMIC_INCREMENT = 2;
+    const RETRY_COUNTER = 10;
+    const BACKOFF_LOGARITHMIC_INCREMENT = 2;
+
+    private int $retryCounter;
+    private int $backoffLogarithmicIncrement;
 
-    public function doIndex(iterable $chunkedCodes, ProgressBar $progressBar, \Closure $codesEsHandler): int
+    /**
+     * BackoffElasticSearchStateHandler constructor.
+     * @param int $retryCounter
+     * @param int $backoffLogarithmicIncrement
+     */
+    public function __construct(int $retryCounter=self::RETRY_COUNTER, int $backoffLogarithmicIncrement=self::BACKOFF_LOGARITHMIC_INCREMENT)
     {
-        $indexedCount = 0;
-
-        $progressBar->start();
-        foreach ($chunkedCodes as $productModelCodes) {
-            $treatedBachSize= sizeof($productModelCodes);
-            $batchSize = sizeof($productModelCodes);
-            $backOverheat = false;
-            $retryCounter = self::RETRY_COUNTER;
-            $waitDelay = self::INITIAL_WAIT_DELAY;
-            $batchEsCodes = $productModelCodes;
-            do {
-                if ($backOverheat) {
-                    echo("Sleeping before retry due to back pressure {$waitDelay} seconds, with batch size of {$batchSize} \n");
-                    sleep($waitDelay);
-                    $batchEsCodes = array_slice($productModelCodes, 0, $batchSize);
-                }
-                try {
-                    $codesEsHandler($batchEsCodes);
-                    array_splice($productModelCodes, 0, $batchSize);
-                    $backOverheat = false;
-                    $retryCounter = self::RETRY_COUNTER;
-                    $waitDelay = self::INITIAL_WAIT_DELAY;
-                    $indexedCount += count($batchEsCodes);
-                } catch (BadRequest400Exception $e) {
-                    if ($e->getCode() == 429) {
-                        $backOverheat = true;
-                        $waitDelay = $waitDelay + self::INITIAL_WAIT_DELAY; //Heuristic: linear increment
-                        $retryCounter--;
-                        $batchSize = intdiv($batchSize, self::BACKOFF_LOGARITHMIC_INCREMENT); //Heuristic: logarithmics decrement
-                    }
-                }
-            } while (($backOverheat && $retryCounter) || sizeof($productModelCodes));
+        $this->retryCounter = $retryCounter;
+        $this->backoffLogarithmicIncrement = $backoffLogarithmicIncrement;
+    }
+
+    protected function resetState(): array
+    {
+        return array(false, $this->retryCounter);
```

Comment:
```suggestion
        return [false, $this->retryCounter];
```

`array` is almost not used anymore